### PR TITLE
Cherry pick user timestamp related PRs

### PR DESCRIPTION
**Pull request by v01dstar**
- https://github.com/facebook/rocksdb/commit/036bbab6f7a0cdeda96609c08a4236c263dd93b3
- https://github.com/facebook/rocksdb/commit/3122cb435875d720fc3d23a48eb7c0fa89d869aa
- https://github.com/facebook/rocksdb/commit/d10c5c08d36a273d7f4122ccbe08c6d98e64488b

### CODE REVIEW

**Comment by tabokie** [comment]
@v01dstar https://github.com/facebook/rocksdb/commit/c9c58a320fb254d5a4fc0cca54fb2221bd79f3cf is not needed. We maintain our own C bindings in rust-rocksdb.

**Comment by tabokie** [comment]
A few other patches that we might need in multi-rocks. Just a reminder. No need to pick them now.

- https://github.com/facebook/rocksdb/commit/95305c44a14ba14fb2233998707cea5f410adc5b
- https://github.com/facebook/rocksdb/commit/16bdb1f999dab165924cb204be9aa8663454c802
- https://github.com/facebook/rocksdb/commit/d4081bf0be585c3853f47915c0b0af72b0a9d1ed
- https://github.com/facebook/rocksdb/commit/8c4ea7b85166d092c4d05d7116172eb8ab9ff484

File: HISTORY.md [hunk 1]
```diff
@@ -21,6 +21,10 @@
 ## 6.29.3 (02/17/2022)
 ### Bug Fixes
 * Fix a data loss bug for 2PC write-committed transaction caused by concurrent transaction commit and memtable switch (#9571).
+### Public API changes
```
Comment (tabokie) [inline HISTORY.md]: Should be put to Unreleased section?

**Review by tabokie — COMMENTED** [review]
Is there any non-trivial modification during the check-pick that I should know of?

File: HISTORY.md (hunk 1 above)
Comment (v01dstar) [reply HISTORY.md]: Done.

**Comment by v01dstar** [comment]
> Is there any non-trivial modification during the check-pick that I should know of?

I didn't resolve any conflicts other than "HISTORY.md"

**Comment by tabokie** [comment]
You might need to re-pick to apply HISTORY.md changes. I'm going to rebase merge so there shouldn't be an extra commit. Rest looks good.

**Review by tabokie — APPROVED** [review]

**Pull request merged**